x-pack/plugins/stack_alerts/server/rule_types/geo_containment/executor.ts

```diff
@@ -51,14 +51,12 @@ export async function executor({
       ? state
       : await getShapeFilters(boundariesRequestMeta, services.scopedClusterClient.asCurrentUser);
 
-  let windowStart = previousStartedAt;
-  if (!windowStart) {
+  if (!previousStartedAt) {
     logger.debug(`alert ${GEO_CONTAINMENT_ID}:${rule.id} alert initialized. Collecting data`);
-    // Consider making first time window configurable?
-    const START_TIME_WINDOW = 1;
-    windowStart = new Date(windowEnd);
-    windowStart.setMinutes(windowStart.getMinutes() - START_TIME_WINDOW);
   }
+  const { dateStart, dateEnd } = getTimeRange('1m');
```

Comment:
This is going to cause problems. For example, assume rule is configured to run every 5 minutes. Using getTimeRange, the rule would only query the past 1 minute and 15 seconds an not the entire time since last running the rule plus 15 seconds. getTimeRange should be passed previousStartAt and base its calculations from that value.

Reply:
This problem is bigger then just the tracking containment rule. This problem will effect any rule where the interval is larger then the value passed to getTimeRange. All rules will miss events that occur within rule interval but after `now - timewindow - config`

Reply:
I think the solution is as simple as `const { dateStart, dateEnd } = getTimeRange(previousStartedAt ? windowEnd - previousStartedAt : '1m');`

Reply:
>All rules will miss events that occur within rule interval but outside of now - timewindow - config

This is by design for most rules (except security solution rules). O11y and other stack rules are generally designed to look at the most recent data (ex: last 5 minutes) on each run in a way that is uncoupled from the rule interval. So if the system experiences delays, instead of falling behind, it misses information in favour of recency. It also helps build consistent time windows for aggregations so the threshold is always over the same period of time.

If the tracking containment rule has varying lookbacks, we can accomodate that with some enhancements. Other rules will mostly work with the following:
- dateStart = now - time window - delay
- dateEnd = now - delay

Reply:
A static lookback window is not going to work for containment alert and would be a breaking change from current behavior.

> This is by design for most rules (except security solution rules)

I would triple check this for stack rule types because if they currently base there lookback off of last run, then this is not going to be the case.

Reply:
Correct, we should revert the code for any rule type that has varying lookbacks until we have made further enhancements. Taking a look at the other stack rules, I did a triple check and can confirm they have static lookbacks before this change:

ES Query:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/stack_alerts/server/rule_types/es_query/executor.ts#L252-L253

Index Threshold:
https://github.com/elastic/kibana/blob/main/x-pack/plugins/stack_alerts/server/rule_types/index_threshold/rule_type.ts#L252-L253

Reply:
For the tracking containment rules, you are right we could change it to `const { dateStart, dateEnd } = getTimeRange(previousStartedAt ? startedAt - previousStartedAt : '1m');`. Would that work for you?